signing.go

```diff
@@ -235,6 +240,10 @@ func (obj JSONWebSignature) Verify(verificationKey interface{}) ([]byte, error)
 		return nil, ErrCryptoFailure
 	}
 
+	if err := verifyCerts(&headers); err != nil {
```

Comment:
We should only verify certificates after we've verified the signature, I don't really trust anything that parses X.509 even if it's written in Go.